spec/mongoid/atomic/modifiers_spec.rb

```diff
@@ -342,6 +342,36 @@
             }
           )
         end
+
+        context "with deep nested conflicting modification" do
+          let(:nested2) do
+            { "addresses.0.locations" => { "number" => "164" } }
+          end
+
+          before do
+            modifiers.push(nested2)
+          end
+
+          it "adds conflicts to the conflicts hash" do
+            expect(modifiers).to eq(
+              { "$push" => {
+                "addresses.0.locations" => { '$each' => [{ "street" => "Bond St" }] } },
+                conflicts: { "$push" =>
+                  { "addresses" => { '$each' => [
+                      { "street" => "Oxford St" }
+                    ] }
+                  },
+                  conflicts: { "$push" =>
```

Comment:
I don't think I understand this. Why are conflicts nested?

Reply:
We have three changes for one association `addresses`:
1. `{ "addresses" => { "street" => "Oxford St" } }`
2. `{ "addresses.0.locations" => { "street" => "Bond St" } }`
3. `{ "addresses.0.locations" => { "number" => "164" } }`
It can't be completed into one query, so in modifiers, we have `:conflict` key, which make the different request. But 2 and 3 changes can't complete into one query too. That's why we use nested conflicts and make the third request. I think this is a very rare case, but anyway.

Reply:
The argument to $each is an array, so I think I expect that that array should include all required attribute sets in it.

Reply:
Why couldn't we do instead:

```
                conflicts: { "$push" =>
                  { "addresses" => { '$each' => [
                      { "street" => "Oxford St" }
                    ] },
                    "addresses.0.locations" => { "$each" => [
                        { "number" => "164" }
                      ] }
                    }
                  }
```

Wouldn't this be equivalent?

Reply:
ok. I relied on this case:
```yaml
{
  "$set"=>
    {"u_at"=>2020-08-16 11:33:09.161392083 UTC,
    "bs.0.u_at"=>2020-08-16 11:33:09.16102952 UTC},
  :conflicts=>
    {"$push"=>
      {"bs.0.cs"=>
        {"$each"=>
          [{"_id"=>BSON::ObjectId('5f39192143c23478de24298e'),
            "c_name"=>"C Name",
            "u_at"=>2020-08-16 11:33:09.160779392 UTC,
            "c_at"=>2020-08-16 11:33:09.160779392 UTC}]},
      "bs"=>
        {"$each"=>
          [{"_id"=>BSON::ObjectId('5f39191043c23478de24298d'),
            "b_name"=>"New B Name",
            "u_at"=>2020-08-16 11:33:09.161229624 UTC,
            "c_at"=>2020-08-16 11:33:09.161229624 UTC}]}}}}
```
There is one `set` and two different `push`. They can't combine into one query. That's why we need to use three requests to complete w/o any errors.
The current test case is forced. I can change `modifiers.push(nested2)` to `modifiers.set(nested2)`. It will be more natural

Reply:
@p see test case here: https://github.com/mongodb/mongoid/pull/4866. This should more clearly illustrate the problem.